server/clustering.go

```diff
@@ -256,6 +252,46 @@ func (s *StanServer) createRaftNode(name string, fsm raft.FSM) (*raftNode, error
 	}, nil
 }
 
+// bootstrapCluster bootstraps the node for the provided Raft group either as a
+// seed node or with the given peer configuration, depending on configuration
+// and with the latter taking precedence.
+func (s *StanServer) bootstrapCluster(name string, node *raft.Raft) error {
```

Comment:
Something look strange here. Since we get in this function with the guarantee that both Bootstrap can't be false and Peers be empty, In both cases you add the current node as the first element. There is no indication when calling `node.BootstrapCluster()` that we are in a bootstrap (seed) mode or not. So if the current code is correct, it looks like you could first do an append of the current node, then if `len(s.opts.Clustering.Peers) > 0` append the peers.
Am I missing something here?

Reply:
Yes, you're right. This was mostly an artifact of some refactoring I did. I will simplify.